modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/change/tracking/spi/resolver/DLFileVersionConstraintResolver.java

```diff
@@ -179,8 +179,14 @@ public void resolveConflict(
 			String oldStoreFileName = entry.getKey();
 			String newStoreFileName = entry.getValue();
 
+			long dlFileEntryPathId = dlFileEntry.getFolderId();
+
+			if (dlFileEntryPathId == 0) {
+				dlFileEntryPathId = dlFileEntry.getRepositoryId();
+			}
+
 			try (InputStream inputStream = DLStoreUtil.getFileAsStream(
-					dlFileEntry.getCompanyId(), dlFileEntry.getRepositoryId(),
+					dlFileEntry.getCompanyId(), dlFileEntryPathId,
```

Comment:
Use getDataRepositoryId; this logic is defined in `dlFileEntry.getDataRepositoryId()`